src/pages/MainPage/GameModeCheckBox.tsx

```diff
@@ -0,0 +1,59 @@
+import * as Checkbox from '@radix-ui/react-checkbox';
+import { CheckIcon } from '@radix-ui/react-icons';
+import { Dispatch, SetStateAction } from 'react';
+import { GAME_MODE_LIST, MAPPED_GAME_MODE_LIST } from './constants/gameMode';
+import { CheckedGameModeType } from './MainPage';
+
+interface GameModeCheckBoxProps {
+  setCheckedGameMode: Dispatch<SetStateAction<CheckedGameModeType>>;
+  checkedGameMode: CheckedGameModeType;
+}
+
+const GameModeCheckBox = ({
+  checkedGameMode,
+  setCheckedGameMode,
+}: GameModeCheckBoxProps) => {
+  return (
+    <form className='flex gap-[3rem]'>
+      {GAME_MODE_LIST.map((mode, i) => {
+        //매 렌더링시 생성되므로 map 바깥으로 뺀다?
+        const onCheckedChange = () => {
+          setCheckedGameMode((prevState) => {
+            const isNewMode = !prevState[mode];
+            if (mode === 'ALL') {
+              const newState = { ...prevState };
+              let checkGameModeState: keyof CheckedGameModeType;
+              for (checkGameModeState in prevState) {
+                newState[checkGameModeState] = isNewMode;
+              }
+              return newState;
+            }
+            return { ...prevState, [mode]: isNewMode };
+          });
+        };
+        return (
+          <div
+            className='flex items-center gap-[0.5em]'
+            key={mode}>
+            <Checkbox.Root
+              checked={checkedGameMode[mode]}
+              onClick={onCheckedChange}
```

Comment:
```suggestion
  const onCheckedChange = (mode: keyof typeof checkedGameMode) => {
    setCheckedGameMode((prevState) => {
      const isNewMode = !prevState[mode];
      if (mode === 'ALL') {
        const newState = { ...prevState };
        let checkGameModeState: keyof CheckedGameModeType;
        for (checkGameModeState in prevState) {
          newState[checkGameModeState] = isNewMode;
        }
        return newState;
      }
      return { ...prevState, [mode]: isNewMode };
    });
  };

  return (
    <form className='flex gap-[3rem]'>
      {GAME_MODE_LIST.map((mode, i) => (
        <div
          className='flex items-center gap-[0.5em]'
          key={mode}>
          <Checkbox.Root
            checked={checkedGameMode[mode]}
            onClick={() => onCheckedChange(mode)}
```
영현님 고민처럼 매 렌더링마다 생성 할 필요는 없을 것 같아요!
이렇게 하면 map 바깥으로 빼도 잘 되더라고요 ㅎㅎ 👍

Reply:
감사합니다 밖으로 빼야겠네요 ㅎㅎ